metrics/metrics-dropwizard/src/main/java/org/apache/james/metrics/dropwizard/DropWizardGaugeRegistry.java

```diff
@@ -25,11 +25,36 @@
 
 import org.apache.james.metrics.api.Gauge;
 import org.apache.james.metrics.api.GaugeRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import com.codahale.metrics.CachedGauge;
 import com.codahale.metrics.MetricRegistry;
 
 public class DropWizardGaugeRegistry implements GaugeRegistry {
+
+    static class ErrorHandlingGauge<T> implements com.codahale.metrics.Gauge<T> {
```

Comment:
I don't understand the goal of this piece of code.
What problem are we trying to fix ?